apps/condo/domains/meter/tasks/sendVerificationDateReminder.js

```diff
@@ -0,0 +1,233 @@
+}
+
+const filterSentReminders = async ({ context, date, reminderWindowSize, remindersPairs }) => {
+    // let's get all user in those reminders
+    const users = uniq(flatten(remindersPairs.map(pair => pair.residents.map(resident => resident.user.id))))
+
+    // now let's retrieve already sent messages by users and message type where statement
+    // filter out messages old that 2 months
+    const sentMessages = await Message.getAll(context, {
+        type: METER_VERIFICATION_DATE_REMINDER_TYPE,
+        user: {
+            id_in: users,
+        },
+        createdAt_gte: dayjs(date).add(-2, 'month').format('YYYY-MM-DD'),
+    })
+
+    // do filter
+    return remindersPairs
+        .map(pair => {
+            const { meter, residents } = pair
+
+            // let's filter residents that already got a notification
+            const residentsWithoutNotifications = residents.filter(resident => {
+                // if we have at least one message
+                // that means we shouldn't send notification again
+                const message = sentMessages.find(sentMessage => {
+                    // we have to check the following things
+                    // 1. Get message for particular resident
+                    // 2. Filter out messages related to other meters
+                    // 3. Filter out messages sent before current reminder window
+                    const messageCreateAt = dayjs(sentMessage.createdAt)
+                    const endOfReminderWindow = dayjs(meter.nextVerificationDate)
+                    const startOfReminderWindow = endOfReminderWindow.add(-reminderWindowSize, 'day')
+
+                    return sentMessage.user.id === resident.user.id
+                        && messageCreateAt.unix() >= startOfReminderWindow.unix()
+                })
+                return message == null
+            })
+
+            // create a pair again
+            // and filter out meter with empty residents
+            return {
+                meter,
+                residents: residentsWithoutNotifications,
+            }
+        })
+        .filter(pair => pair.residents.length > 0)
+}
+
+const getOrganizationLang = async (context, id) => {
+    const organization = await Organization.getOne(context, { id, deletedAt: null })
+
+    /**
+     * Detect message language
+     * Use DEFAULT_LOCALE if organization.country is unknown
+     * (not defined within @condo/domains/common/constants/countries)
+     */
+    return get(COUNTRIES, get(organization, 'country.locale'), DEFAULT_LOCALE)
+}
+const generateReminderMessages = async ({ context, reminderWindowSize, reminders }) => {
+    const messages = []
+
+    await Promise.all(reminders.map(async (reminder) => {
+        const { meter, residents } = reminder
+        const lang = await getOrganizationLang(context, meter.organization.id)
+
+        // prepare a message for each resident
+        messages.push(
+            ...residents.map(resident => ({
```

Comment:
In case if I omit destructuring here, the list from [object, object] become [[object], [object]]
To avoid complex Array in Array structure I desctructuring items before put them into the collection